From 6ecf78bdda5d863cd4b6c04d8d1e2fea258296b3 Mon Sep 17 00:00:00 2001
From: Aleksey Kladov <aleksey.kladov@gmail.com>
Date: Thu, 22 Sep 2022 17:49:39 +0100
Subject: [PATCH 1/2] chore: upgrade Rust to 1.64.0

https://blog.rust-lang.org/2022/09/22/Rust-1.64.0.html

Notable changes:

* rust-analyzer is now available via rustup :o)
* cargo supports "workspace inheritance". We can use this to cut some
  repeated deps and avoid repating rust-version everywhere. I'll handle
  that in a separate PR
* poll_fn and ready! are stabilized, we might use them for some of our
  async stuff
* add Nintendo Switch as tier 3 target so I guess we can run neard on
  switch now?
---
 chain/chain-primitives/Cargo.toml                             | 2 +-
 chain/chunks-primitives/Cargo.toml                            | 2 +-
 chain/chunks/Cargo.toml                                       | 2 +-
 chain/client-primitives/Cargo.toml                            | 2 +-
 chain/client/Cargo.toml                                       | 2 +-
 chain/epoch-manager/Cargo.toml                                | 2 +-
 chain/indexer-primitives/Cargo.toml                           | 2 +-
 chain/indexer/Cargo.toml                                      | 2 +-
 chain/jsonrpc-adversarial-primitives/Cargo.toml               | 2 +-
 chain/jsonrpc-primitives/Cargo.toml                           | 2 +-
 chain/jsonrpc/Cargo.toml                                      | 2 +-
 chain/jsonrpc/client/Cargo.toml                               | 2 +-
 chain/jsonrpc/fuzz/Cargo.toml                                 | 2 +-
 chain/jsonrpc/jsonrpc-tests/Cargo.toml                        | 2 +-
 chain/network-primitives/Cargo.toml                           | 2 +-
 chain/network/Cargo.toml                                      | 2 +-
 chain/pool/Cargo.toml                                         | 2 +-
 chain/rosetta-rpc/Cargo.toml                                  | 2 +-
 chain/telemetry/Cargo.toml                                    | 2 +-
 core/account-id/fuzz/Cargo.toml                               | 2 +-
 core/chain-configs/Cargo.toml                                 | 2 +-
 core/crypto/Cargo.toml                                        | 2 +-
 core/o11y/Cargo.toml                                          | 2 +-
 core/primitives-core/Cargo.toml                               | 2 +-
 core/primitives/Cargo.toml                                    | 2 +-
 core/store/Cargo.toml                                         | 2 +-
 genesis-tools/genesis-csv-to-json/Cargo.toml                  | 2 +-
 genesis-tools/genesis-populate/Cargo.toml                     | 2 +-
 genesis-tools/keypair-generator/Cargo.toml                    | 2 +-
 integration-tests/Cargo.toml                                  | 2 +-
 nearcore/Cargo.toml                                           | 2 +-
 neard/Cargo.toml                                              | 2 +-
 runtime/near-test-contracts/Cargo.toml                        | 2 +-
 .../near-test-contracts/contract-for-fuzzing-rs/Cargo.toml    | 2 +-
 runtime/near-test-contracts/estimator-contract/Cargo.toml     | 2 +-
 runtime/near-test-contracts/test-contract-rs/Cargo.toml       | 2 +-
 runtime/near-vm-errors/Cargo.toml                             | 2 +-
 runtime/near-vm-logic/Cargo.toml                              | 2 +-
 runtime/near-vm-runner/Cargo.toml                             | 4 ++--
 runtime/near-vm-runner/fuzz/Cargo.toml                        | 2 +-
 runtime/runtime-params-estimator/Cargo.toml                   | 4 ++--
 runtime/runtime-params-estimator/emu-cost/Dockerfile          | 2 +-
 .../runtime-params-estimator/estimator-warehouse/Cargo.toml   | 2 +-
 runtime/runtime/Cargo.toml                                    | 2 +-
 rust-toolchain.toml                                           | 2 +-
 test-utils/actix-test-utils/Cargo.toml                        | 2 +-
 test-utils/logger/Cargo.toml                                  | 2 +-
 test-utils/runtime-tester/Cargo.toml                          | 2 +-
 test-utils/runtime-tester/fuzz/Cargo.toml                     | 2 +-
 test-utils/store-validator/Cargo.toml                         | 2 +-
 test-utils/testlib/Cargo.toml                                 | 2 +-
 tools/chainsync-loadtest/Cargo.toml                           | 2 +-
 tools/delay-detector/Cargo.toml                               | 2 +-
 tools/indexer/example/Cargo.toml                              | 2 +-
 tools/mock-node/Cargo.toml                                    | 2 +-
 tools/restaked/Cargo.toml                                     | 2 +-
 tools/rpctypegen/core/Cargo.toml                              | 2 +-
 tools/rpctypegen/macro/Cargo.toml                             | 2 +-
 tools/state-viewer/Cargo.toml                                 | 2 +-
 tools/storage-usage-delta-calculator/Cargo.toml               | 2 +-
 tools/themis/Cargo.toml                                       | 2 +-
 utils/mainnet-res/Cargo.toml                                  | 2 +-
 utils/near-cache/Cargo.toml                                   | 2 +-
 utils/near-performance-metrics-macros/Cargo.toml              | 2 +-
 utils/near-performance-metrics/Cargo.toml                     | 2 +-
 utils/near-stable-hasher/Cargo.toml                           | 2 +-
 66 files changed, 68 insertions(+), 68 deletions(-)

diff --git a/chain/chain-primitives/Cargo.toml b/chain/chain-primitives/Cargo.toml
index bdf0fa15a5..7bee072bc6 100644
--- a/chain/chain-primitives/Cargo.toml
+++ b/chain/chain-primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/chain/chunks-primitives/Cargo.toml b/chain/chunks-primitives/Cargo.toml
index be35f5e139..bb8be135aa 100644
--- a/chain/chunks-primitives/Cargo.toml
+++ b/chain/chunks-primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/chain/chunks/Cargo.toml b/chain/chunks/Cargo.toml
index 3278919cf5..da4018c12c 100644
--- a/chain/chunks/Cargo.toml
+++ b/chain/chunks/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/client-primitives/Cargo.toml b/chain/client-primitives/Cargo.toml
index 6958db2463..7d3d8220bb 100644
--- a/chain/client-primitives/Cargo.toml
+++ b/chain/client-primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/chain/client/Cargo.toml b/chain/client/Cargo.toml
index 1eda63b931..57a074942b 100644
--- a/chain/client/Cargo.toml
+++ b/chain/client/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/epoch-manager/Cargo.toml b/chain/epoch-manager/Cargo.toml
index 5a3913372f..dd638f75ad 100644
--- a/chain/epoch-manager/Cargo.toml
+++ b/chain/epoch-manager/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 
diff --git a/chain/indexer-primitives/Cargo.toml b/chain/indexer-primitives/Cargo.toml
index eb85531cb6..45b2b6d57d 100644
--- a/chain/indexer-primitives/Cargo.toml
+++ b/chain/indexer-primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/chain/indexer/Cargo.toml b/chain/indexer/Cargo.toml
index ab390b89f3..0133cdb37a 100644
--- a/chain/indexer/Cargo.toml
+++ b/chain/indexer/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/jsonrpc-adversarial-primitives/Cargo.toml b/chain/jsonrpc-adversarial-primitives/Cargo.toml
index b2622f1841..b6f2f97ae3 100644
--- a/chain/jsonrpc-adversarial-primitives/Cargo.toml
+++ b/chain/jsonrpc-adversarial-primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/jsonrpc-primitives/Cargo.toml b/chain/jsonrpc-primitives/Cargo.toml
index 07f92ff901..819c9208ab 100644
--- a/chain/jsonrpc-primitives/Cargo.toml
+++ b/chain/jsonrpc-primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/chain/jsonrpc/Cargo.toml b/chain/jsonrpc/Cargo.toml
index ee0842c825..35c7611e06 100644
--- a/chain/jsonrpc/Cargo.toml
+++ b/chain/jsonrpc/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/jsonrpc/client/Cargo.toml b/chain/jsonrpc/client/Cargo.toml
index 101b99f0a9..ff8bd08e7a 100644
--- a/chain/jsonrpc/client/Cargo.toml
+++ b/chain/jsonrpc/client/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/jsonrpc/fuzz/Cargo.toml b/chain/jsonrpc/fuzz/Cargo.toml
index 85eb5a5cd7..0c22049c7f 100644
--- a/chain/jsonrpc/fuzz/Cargo.toml
+++ b/chain/jsonrpc/fuzz/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [package.metadata]
diff --git a/chain/jsonrpc/jsonrpc-tests/Cargo.toml b/chain/jsonrpc/jsonrpc-tests/Cargo.toml
index d729deb98b..b1dd2c9961 100644
--- a/chain/jsonrpc/jsonrpc-tests/Cargo.toml
+++ b/chain/jsonrpc/jsonrpc-tests/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/network-primitives/Cargo.toml b/chain/network-primitives/Cargo.toml
index 7ce8507100..725efaa444 100644
--- a/chain/network-primitives/Cargo.toml
+++ b/chain/network-primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2021"
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 description = "This crate hosts NEAR network-related primitive types"
 repository = "https://github.com/near/nearcore"
 license = "MIT OR Apache-2.0"
diff --git a/chain/network/Cargo.toml b/chain/network/Cargo.toml
index 8b08768bab..b9810ecedb 100644
--- a/chain/network/Cargo.toml
+++ b/chain/network/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2021"
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 publish = false
 
 [build-dependencies]
diff --git a/chain/pool/Cargo.toml b/chain/pool/Cargo.toml
index f72945b66b..8db9d2d237 100644
--- a/chain/pool/Cargo.toml
+++ b/chain/pool/Cargo.toml
@@ -5,7 +5,7 @@ license = "MIT OR Apache-2.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/chain/rosetta-rpc/Cargo.toml b/chain/rosetta-rpc/Cargo.toml
index 48ed319a89..56ab93d06e 100644
--- a/chain/rosetta-rpc/Cargo.toml
+++ b/chain/rosetta-rpc/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
diff --git a/chain/telemetry/Cargo.toml b/chain/telemetry/Cargo.toml
index 57601cc004..9c606ca9c8 100644
--- a/chain/telemetry/Cargo.toml
+++ b/chain/telemetry/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/core/account-id/fuzz/Cargo.toml b/core/account-id/fuzz/Cargo.toml
index 729bb04a5c..51a206135d 100644
--- a/core/account-id/fuzz/Cargo.toml
+++ b/core/account-id/fuzz/Cargo.toml
@@ -5,7 +5,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [package.metadata]
diff --git a/core/chain-configs/Cargo.toml b/core/chain-configs/Cargo.toml
index 855365bea2..a24a6253a2 100644
--- a/core/chain-configs/Cargo.toml
+++ b/core/chain-configs/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/core/crypto/Cargo.toml b/core/crypto/Cargo.toml
index 5a631bcf3b..b6024cc1fa 100644
--- a/core/crypto/Cargo.toml
+++ b/core/crypto/Cargo.toml
@@ -5,7 +5,7 @@ license = "MIT OR Apache-2.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 repository = "https://github.com/near/nearcore"
 description = "This is an internal crate for common cryptographic types"
diff --git a/core/o11y/Cargo.toml b/core/o11y/Cargo.toml
index 05c0839cb4..5c23aecaf6 100644
--- a/core/o11y/Cargo.toml
+++ b/core/o11y/Cargo.toml
@@ -6,7 +6,7 @@ description = "Observability helpers for the near codebase"
 edition = "2021"
 publish = false
 readme = "README.md"
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 
 [dependencies]
 atty = "0.2"
diff --git a/core/primitives-core/Cargo.toml b/core/primitives-core/Cargo.toml
index 079c18f090..754ab9e3f7 100644
--- a/core/primitives-core/Cargo.toml
+++ b/core/primitives-core/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/core/primitives/Cargo.toml b/core/primitives/Cargo.toml
index 701268f4d6..af6903145b 100644
--- a/core/primitives/Cargo.toml
+++ b/core/primitives/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/core/store/Cargo.toml b/core/store/Cargo.toml
index 7171b86f0f..6821b0178b 100644
--- a/core/store/Cargo.toml
+++ b/core/store/Cargo.toml
@@ -5,7 +5,7 @@ license = "MIT OR Apache-2.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/genesis-tools/genesis-csv-to-json/Cargo.toml b/genesis-tools/genesis-csv-to-json/Cargo.toml
index 30848a8e68..7423ac9c3e 100644
--- a/genesis-tools/genesis-csv-to-json/Cargo.toml
+++ b/genesis-tools/genesis-csv-to-json/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/genesis-tools/genesis-populate/Cargo.toml b/genesis-tools/genesis-populate/Cargo.toml
index 5883729110..fb3bd33f53 100644
--- a/genesis-tools/genesis-populate/Cargo.toml
+++ b/genesis-tools/genesis-populate/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/genesis-tools/keypair-generator/Cargo.toml b/genesis-tools/keypair-generator/Cargo.toml
index 6b417f4708..d0210e930f 100644
--- a/genesis-tools/keypair-generator/Cargo.toml
+++ b/genesis-tools/keypair-generator/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/integration-tests/Cargo.toml b/integration-tests/Cargo.toml
index 6b71a1a0f3..6aff950b17 100644
--- a/integration-tests/Cargo.toml
+++ b/integration-tests/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/nearcore/Cargo.toml b/nearcore/Cargo.toml
index 15e374d274..8f9813c7cd 100644
--- a/nearcore/Cargo.toml
+++ b/nearcore/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/neard/Cargo.toml b/neard/Cargo.toml
index cd2a8700da..9d4853ce09 100644
--- a/neard/Cargo.toml
+++ b/neard/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 default-run = "neard"
 
diff --git a/runtime/near-test-contracts/Cargo.toml b/runtime/near-test-contracts/Cargo.toml
index 07534dece9..1cdda579ab 100644
--- a/runtime/near-test-contracts/Cargo.toml
+++ b/runtime/near-test-contracts/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 publish = false
 authors = ["Near Inc <hello@nearprotocol.com>"]
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "Apache-2.0"
 
diff --git a/runtime/near-test-contracts/contract-for-fuzzing-rs/Cargo.toml b/runtime/near-test-contracts/contract-for-fuzzing-rs/Cargo.toml
index 566d3283c0..1c42d02b90 100644
--- a/runtime/near-test-contracts/contract-for-fuzzing-rs/Cargo.toml
+++ b/runtime/near-test-contracts/contract-for-fuzzing-rs/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.1.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2018"
 
 [lib]
diff --git a/runtime/near-test-contracts/estimator-contract/Cargo.toml b/runtime/near-test-contracts/estimator-contract/Cargo.toml
index c516edcbc2..3c7db05674 100644
--- a/runtime/near-test-contracts/estimator-contract/Cargo.toml
+++ b/runtime/near-test-contracts/estimator-contract/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.1.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [lib]
diff --git a/runtime/near-test-contracts/test-contract-rs/Cargo.toml b/runtime/near-test-contracts/test-contract-rs/Cargo.toml
index b080a45c5c..01f3e05797 100644
--- a/runtime/near-test-contracts/test-contract-rs/Cargo.toml
+++ b/runtime/near-test-contracts/test-contract-rs/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.1.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [lib]
diff --git a/runtime/near-vm-errors/Cargo.toml b/runtime/near-vm-errors/Cargo.toml
index 971a9cc553..311046c16a 100644
--- a/runtime/near-vm-errors/Cargo.toml
+++ b/runtime/near-vm-errors/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 readme = "README.md"
diff --git a/runtime/near-vm-logic/Cargo.toml b/runtime/near-vm-logic/Cargo.toml
index e75f190e62..9cf6e08bf6 100644
--- a/runtime/near-vm-logic/Cargo.toml
+++ b/runtime/near-vm-logic/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 readme = "README.md"
diff --git a/runtime/near-vm-runner/Cargo.toml b/runtime/near-vm-runner/Cargo.toml
index ee15cd09e6..305b06252a 100644
--- a/runtime/near-vm-runner/Cargo.toml
+++ b/runtime/near-vm-runner/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 readme = "README.md"
@@ -109,7 +109,7 @@ io_trace = ["near-vm-logic/io_trace"]
 protocol_feature_ed25519_verify = [
     "near-primitives/protocol_feature_ed25519_verify",
     "near-vm-logic/protocol_feature_ed25519_verify"
-] 
+]
 
 [package.metadata.cargo-udeps.ignore]
 # `no_cache` feature leads to an unused `cached` crate
diff --git a/runtime/near-vm-runner/fuzz/Cargo.toml b/runtime/near-vm-runner/fuzz/Cargo.toml
index 2181ceeb7d..83e7fad0f8 100644
--- a/runtime/near-vm-runner/fuzz/Cargo.toml
+++ b/runtime/near-vm-runner/fuzz/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [package.metadata]
diff --git a/runtime/runtime-params-estimator/Cargo.toml b/runtime/runtime-params-estimator/Cargo.toml
index 4555d6cba9..1c8c395c96 100644
--- a/runtime/runtime-params-estimator/Cargo.toml
+++ b/runtime/runtime-params-estimator/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [[bin]]
@@ -80,4 +80,4 @@ io_trace = ["near-store/io_trace", "near-o11y/io_trace", "near-vm-logic/io_trace
 protocol_feature_ed25519_verify = [
     "near-vm-logic/protocol_feature_ed25519_verify",
     "near-vm-runner/protocol_feature_ed25519_verify"
-]
\ No newline at end of file
+]
diff --git a/runtime/runtime-params-estimator/emu-cost/Dockerfile b/runtime/runtime-params-estimator/emu-cost/Dockerfile
index 2b67283740..9966e18b9e 100644
--- a/runtime/runtime-params-estimator/emu-cost/Dockerfile
+++ b/runtime/runtime-params-estimator/emu-cost/Dockerfile
@@ -1,5 +1,5 @@
 # our local base image
-FROM rust:1.63.0
+FROM rust:1.64.0
 
 LABEL description="Container for builds"
 
diff --git a/runtime/runtime-params-estimator/estimator-warehouse/Cargo.toml b/runtime/runtime-params-estimator/estimator-warehouse/Cargo.toml
index 8b8c2a43fa..d53d867e62 100644
--- a/runtime/runtime-params-estimator/estimator-warehouse/Cargo.toml
+++ b/runtime/runtime-params-estimator/estimator-warehouse/Cargo.toml
@@ -4,7 +4,7 @@ name = "estimator-warehouse"
 version = "0.0.0"
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/runtime/runtime/Cargo.toml b/runtime/runtime/Cargo.toml
index f82b7b7055..be5c0715b1 100644
--- a/runtime/runtime/Cargo.toml
+++ b/runtime/runtime/Cargo.toml
@@ -5,7 +5,7 @@ license = "MIT OR Apache-2.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/rust-toolchain.toml b/rust-toolchain.toml
index 7795a53e4a..ecc4031d82 100644
--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -2,6 +2,6 @@
 # This specifies the version of Rust we use to build.
 # Individual crates in the workspace may support a lower version, as indicated by `rust-version` field in each crate's `Cargo.toml`.
 # The version specified below, should be at least as high as the maximum `rust-version` within the workspace.
-channel = "1.63.0"
+channel = "1.64.0"
 components = [ "rustfmt" ]
 targets = [ "wasm32-unknown-unknown" ]
diff --git a/test-utils/actix-test-utils/Cargo.toml b/test-utils/actix-test-utils/Cargo.toml
index 450a0f4c82..8fd47568ca 100644
--- a/test-utils/actix-test-utils/Cargo.toml
+++ b/test-utils/actix-test-utils/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/test-utils/logger/Cargo.toml b/test-utils/logger/Cargo.toml
index 4a6588bb1d..655419bb87 100644
--- a/test-utils/logger/Cargo.toml
+++ b/test-utils/logger/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/test-utils/runtime-tester/Cargo.toml b/test-utils/runtime-tester/Cargo.toml
index 11ee21b958..d551616020 100644
--- a/test-utils/runtime-tester/Cargo.toml
+++ b/test-utils/runtime-tester/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/test-utils/runtime-tester/fuzz/Cargo.toml b/test-utils/runtime-tester/fuzz/Cargo.toml
index 0196c9875e..25bedfc26b 100644
--- a/test-utils/runtime-tester/fuzz/Cargo.toml
+++ b/test-utils/runtime-tester/fuzz/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [package.metadata]
diff --git a/test-utils/store-validator/Cargo.toml b/test-utils/store-validator/Cargo.toml
index 5c230e8d6c..7e170ba37c 100644
--- a/test-utils/store-validator/Cargo.toml
+++ b/test-utils/store-validator/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/test-utils/testlib/Cargo.toml b/test-utils/testlib/Cargo.toml
index e968fc03d0..5190abf978 100644
--- a/test-utils/testlib/Cargo.toml
+++ b/test-utils/testlib/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/tools/chainsync-loadtest/Cargo.toml b/tools/chainsync-loadtest/Cargo.toml
index a66ef741b2..26b500adef 100644
--- a/tools/chainsync-loadtest/Cargo.toml
+++ b/tools/chainsync-loadtest/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 default-run = "chainsync-loadtest"
 
diff --git a/tools/delay-detector/Cargo.toml b/tools/delay-detector/Cargo.toml
index 23c8ae72a2..3014f33d00 100644
--- a/tools/delay-detector/Cargo.toml
+++ b/tools/delay-detector/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/tools/indexer/example/Cargo.toml b/tools/indexer/example/Cargo.toml
index 6d1bf78004..4d8441f57b 100644
--- a/tools/indexer/example/Cargo.toml
+++ b/tools/indexer/example/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/tools/mock-node/Cargo.toml b/tools/mock-node/Cargo.toml
index 5f099fb4ac..83b98ee8e5 100644
--- a/tools/mock-node/Cargo.toml
+++ b/tools/mock-node/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/tools/restaked/Cargo.toml b/tools/restaked/Cargo.toml
index 45e7518c32..51ef864f1e 100644
--- a/tools/restaked/Cargo.toml
+++ b/tools/restaked/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/tools/rpctypegen/core/Cargo.toml b/tools/rpctypegen/core/Cargo.toml
index 14a01e42a9..5121924c5c 100644
--- a/tools/rpctypegen/core/Cargo.toml
+++ b/tools/rpctypegen/core/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/tools/rpctypegen/macro/Cargo.toml b/tools/rpctypegen/macro/Cargo.toml
index ef1bcc29b6..d00b698509 100644
--- a/tools/rpctypegen/macro/Cargo.toml
+++ b/tools/rpctypegen/macro/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = true
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/near/nearcore"
diff --git a/tools/state-viewer/Cargo.toml b/tools/state-viewer/Cargo.toml
index 6edc72c65e..c05ce4ad7d 100644
--- a/tools/state-viewer/Cargo.toml
+++ b/tools/state-viewer/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 [dependencies]
diff --git a/tools/storage-usage-delta-calculator/Cargo.toml b/tools/storage-usage-delta-calculator/Cargo.toml
index 4925635ab6..dcb3657c90 100644
--- a/tools/storage-usage-delta-calculator/Cargo.toml
+++ b/tools/storage-usage-delta-calculator/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>", "Egor Kulikov <egor@near.org>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
diff --git a/tools/themis/Cargo.toml b/tools/themis/Cargo.toml
index 7c948f3be7..3cf12d1f1c 100644
--- a/tools/themis/Cargo.toml
+++ b/tools/themis/Cargo.toml
@@ -3,7 +3,7 @@ name = "themis"
 version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>", "Miraculous Owonubi <omiraculous@gmail.com>"]
 edition = "2021"
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 publish = false
 license = "MIT OR Apache-2.0"
 description = "https://en.wikipedia.org/wiki/Themis"
diff --git a/utils/mainnet-res/Cargo.toml b/utils/mainnet-res/Cargo.toml
index 4154869e7a..efe65e24f3 100644
--- a/utils/mainnet-res/Cargo.toml
+++ b/utils/mainnet-res/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2021"
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 publish = false
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
diff --git a/utils/near-cache/Cargo.toml b/utils/near-cache/Cargo.toml
index 5ac2b986b1..bbbddc1dc3 100644
--- a/utils/near-cache/Cargo.toml
+++ b/utils/near-cache/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2021"
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 publish = false
 
 [dependencies]
diff --git a/utils/near-performance-metrics-macros/Cargo.toml b/utils/near-performance-metrics-macros/Cargo.toml
index a88e0bdec4..05051201af 100644
--- a/utils/near-performance-metrics-macros/Cargo.toml
+++ b/utils/near-performance-metrics-macros/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2021"
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 publish = false
 
 [dependencies]
diff --git a/utils/near-performance-metrics/Cargo.toml b/utils/near-performance-metrics/Cargo.toml
index 62c2da77d7..71dc4ec462 100644
--- a/utils/near-performance-metrics/Cargo.toml
+++ b/utils/near-performance-metrics/Cargo.toml
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2021"
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 publish = false
 
 [dependencies]
diff --git a/utils/near-stable-hasher/Cargo.toml b/utils/near-stable-hasher/Cargo.toml
index 9f88cf7b96..0d6bb783b4 100644
--- a/utils/near-stable-hasher/Cargo.toml
+++ b/utils/near-stable-hasher/Cargo.toml
@@ -4,5 +4,5 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2021"
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.63.0"
+rust-version = "1.64.0"
 publish = false

From 03a078ee4463d11f165c09b5fd7cfbea69147188 Mon Sep 17 00:00:00 2001
From: Aleksey Kladov <aleksey.kladov@gmail.com>
Date: Thu, 22 Sep 2022 17:54:57 +0100
Subject: [PATCH 2/2] fix new warnings

---
 runtime/near-vm-runner/src/wasmtime_runner.rs   |  2 +-
 .../src/function_call.rs                        | 17 +++++------------
 2 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/runtime/near-vm-runner/src/wasmtime_runner.rs b/runtime/near-vm-runner/src/wasmtime_runner.rs
index ab3d173156..4a0256860c 100644
--- a/runtime/near-vm-runner/src/wasmtime_runner.rs
+++ b/runtime/near-vm-runner/src/wasmtime_runner.rs
@@ -283,7 +283,7 @@ impl crate::runner::VM for WasmtimeVM {
                 Some(func) => match func.typed::<(), (), _>(&mut store) {
                     Ok(run) => match run.call(&mut store, ()) {
                         Ok(_) => VMResult::ok(logic),
-                        Err(err) => (VMResult::abort(logic, err.into_vm_error())),
+                        Err(err) => VMResult::abort(logic, err.into_vm_error()),
                     },
                     Err(err) => VMResult::abort(logic, err.into_vm_error()),
                 },
diff --git a/runtime/runtime-params-estimator/src/function_call.rs b/runtime/runtime-params-estimator/src/function_call.rs
index 145dd87173..0b662a2674 100644
--- a/runtime/runtime-params-estimator/src/function_call.rs
+++ b/runtime/runtime-params-estimator/src/function_call.rs
@@ -39,26 +39,19 @@ fn make_many_methods_contract(method_count: i32) -> ContractCode {
     for i in 0..method_count {
         write!(
             &mut methods,
-            "
-            (export \"hello{}\" (func {i}))
-              (func (;{i};)
+            r#"
+            (export "hello{i}" (func {i}))
+              (func
                 i32.const {i}
                 drop
                 return
               )
-            ",
-            i = i
+            "#,
         )
         .unwrap();
     }
 
-    let code = format!(
-        "
-        (module
-            {}
-            )",
-        methods
-    );
+    let code = format!("(module {methods})");
     ContractCode::new(wat::parse_str(code).unwrap(), None)
 }
 
